### ISSUE TITLE
[RN] onDeviceUI of getStorybookUI cause a bug with action addon and fast refresh

### ISSUE THREAD

**Issue by JeffGuKang** [storybookjs/storybook#8441]
**Describe the bug**
I set `onDeviceUI` to false to avoid #8274.

But `onDeviceUI: false` causes error message `TypeError: JSON.stringify cannot serialize` when use `action` addons or run fast refresh.

```
const StorybookUIRoot = getStorybookUI({
  onDeviceUI: false,
  host: hostname,
  shouldPersistSelection: true,
})
```

React Native > 0.61
Storybook packages > 5.2.1

**To Reproduce**

1. Turn on storybook server with react native.
2. Trigger action addon

**Expected behavior**
Trigger action and show on addon page.

**Screenshots**

<img width="326" alt="image" src="https://user-images.githubusercontent.com/216363/66918834-2716b300-f05b-11e9-9ef6-38316f7b99a8.png">

**Code snippets**
If applicable, add code samples to help explain your problem.

**System:**
```
System:
    OS: macOS Mojave 10.14.6
    CPU: (16) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
  Binaries:
    Node: 10.16.0 - ~/.nvm/versions/node/v10.16.0/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.16.0/bin/npm
  Browsers:
    Chrome: 77.0.3865.120
    Firefox: 69.0.3
    Safari: 13.0.1
  npmPackages:
    @storybook/addon-actions: v5.3.0-alpha.6 => 5.3.0-alpha.6 
    @storybook/addon-knobs: v5.3.0-alpha.6 => 5.3.0-alpha.6 
    @storybook/addon-links: v5.3.0-alpha.6 => 5.3.0-alpha.6 
    @storybook/addon-ondevice-actions: v5.3.0-alpha.6 => 5.3.0-alpha.6 
    @storybook/addon-ondevice-knobs: v5.3.0-alpha.6 => 5.3.0-alpha.6 
    @storybook/addons: v5.3.0-alpha.6 => 5.3.0-alpha.6 
    @storybook/react-native: v5.3.0-alpha.6 => 5.3.0-alpha.6 
    @storybook/react-native-server: v5.3.0-alpha.6 => 5.3.0-alpha.6 
```

**Comment by JeffGuKang** [comment]
#8274 was fixed in #8681.

**Comment by Gongreg** [comment]
Hi.

There are multiple issues you are listing here.

https://github.com/storybookjs/storybook/issues/6471 (actions addon issue, can be avoided by not using react events in action:

```javascript
<Button onPress={() => action('Some event')()}/>
```

https://github.com/storybookjs/storybook/issues/7916 issue Regarding Fast refresh.

I am closing this one in favour of those two.
